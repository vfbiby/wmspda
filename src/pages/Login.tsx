import React from 'react';
import {Text, View, TextInput, Button, Alert} from 'react-native';
import {useForm, Controller} from 'react-hook-form';

interface ILoginForm {
  username: string;
  password: string;
}

const Login = () => {
  const {control, handleSubmit, errors} = useForm();
  const onSubmit = (data: ILoginForm) => {
    Alert.alert(data.username);
    console.log(data);
  };

  return (
    <View>
      <Controller
        control={control}
        render={({onChange, onBlur, value}) => (
          <TextInput
            onBlur={onBlur}
            placeholder="Username"
            onChangeText={(value) => onChange(value)}
            value={value}
          />
        )}
        name="username"
        rules={{required: true}}
        defaultValue=""
      />
      {errors.username && <Text>Username is required!</Text>}

      <Controller
        control={control}
        render={({onChange, onBlur, value}) => (
          <TextInput
            onBlur={onBlur}
            placeholder="Password"
            onChangeText={(value) => onChange(value)}
            value={value}
          />
        )}
        name="password"
        defaultValue=""
      />

      <Button title="Submit" onPress={handleSubmit(onSubmit)} />
    </View>
  );
};

export default Login;
