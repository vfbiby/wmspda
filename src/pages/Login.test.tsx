import {fireEvent, render, waitFor} from '@testing-library/react-native';
import React from 'react';
import Login from './Login';

describe('Login page', () => {
  it('should show username and password textinput with no errors when first render', () => {
    const {getByPlaceholderText, queryByText} = render(<Login />);
    expect(getByPlaceholderText('Username')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(queryByText('Username is required!')).toBeFalsy();
    expect(queryByText('Password is required!')).toBeFalsy();
  });

  it('should display required error while username is invalid', async () => {
    //Given we have a form
    const {getByPlaceholderText, getByText, toJSON} = render(<Login />);
    //When we don't type a username and click login button
    fireEvent.changeText(getByPlaceholderText('Password'), 'password');
    //Then an alert should display to tell us to type a username for logining
    fireEvent.press(getByText('Submit'));

    await waitFor(() =>
      expect(getByText('Username is required!')).toBeTruthy(),
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
