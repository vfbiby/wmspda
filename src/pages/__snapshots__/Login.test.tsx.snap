// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should display required error while username is invalid 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />
  <Text>
    Username is required!
  </Text>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value="password"
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
